src/components/UnderlyingToken.tsx

```diff
@@ -43,19 +45,22 @@ export const UnderlyingToken = ({ assetId }: UnderlyingTokenProps) => {
     ;(async () => {
       try {
         if (shouldHide || !wallet) return
+        moduleLogger.trace({ asset, fn: 'yearn.token' }, 'Get Yearn Token')
         const token = await yearn.token({ vaultAddress: asset.tokenId! })
         const chain = asset.chain
         const network = NetworkTypes.MAINNET
         const assetNamespace = AssetNamespace.ERC20
         const assetReference = toLower(token)
-        setUnderlyingCAIP19(caip19.toCAIP19({ chain, network, assetNamespace, assetReference }))
+        const assetId = caip19.toCAIP19({ chain, network, assetNamespace, assetReference })
+        moduleLogger.trace({ assetId, fn: 'yearn.token' }, 'Yearn Asset')
+        setUnderlyingAssetId(assetId)
       } catch (error) {
-        console.error(error)
+        moduleLogger.error(error, 'yearn.token() failed')
       }
     })()
   }, [shouldHide, asset.tokenId, asset.chain, vault, wallet, yearn])
```

Comment:
Asset is now a dependency of this `useEffect`.

```suggestion
  }, [shouldHide, asset.tokenId, asset.chain, vault, wallet, yearn, asset])
```

Reply:
I was trying to avoid making any changes that weren't directly logging related to keep the risk low.